Sort Matrix exercise

anciuvaga opened the pull request.

anciuvaga committed:
Adding sort matrix exercise

CristianManic commented on src/javaExercises/SortMatrix/SortMatrix.java:
```diff
@@ -0,0 +1,85 @@
+        }
+    }
+
+    public int[] sortLines(int[][] arr) {
```
try using a Map<Integer,Integer[]>, were key will be the sum of elements per line and value will be the line from array, then you can just sort keys from map and get lines in sorted order of the keys

anciuvaga replied:
Done

CristianManic commented on src/javaExercises/SortMatrix/SortMatrix.java:
```diff
@@ -0,0 +1,85 @@
+
+public class SortMatrix {
+    public int[][] populateArrayWithRandomValues(int n, int max, int min) {
+
```
don't leave empty lines here

anciuvaga replied:
Done

CristianManic commented on src/javaExercises/SortMatrix/SortMatrix.java:
```diff
@@ -0,0 +1,85 @@
+import java.util.Random;
+
+public class SortMatrix {
+    public int[][] populateArrayWithRandomValues(int n, int max, int min) {
```
1. give a more meaningful name to your variables
2. I think that min and max variables are redundant here, just set some basic values

anciuvaga replied:
Done

anciuvaga committed:
Adding a Map to get lines in sorted order of the keys

CristianManic commented on src/javaExercises/SortMatrix/SortMatrix.java:
```diff
@@ -0,0 +1,34 @@
+        System.out.println("Unsorted Matrix:");
+        sortMatrix.printArray(arrMap);
+        System.out.println("Sorted Matrix:");
+        sortMatrix.printArray(new TreeMap<Integer, Integer[]>(arrMap));
```
for practice and more understanding how this staff work, do not use ***TreeMap***, make you own implementation for sorting

anciuvaga replied:
Done

anciuvaga committed:
Modifying Map sorting implementation

anciuvaga committed:
Extract keys from map
Adding bouble sort method to sort the keys
Print the map's value based on sorted keys
